react/src/pages/utils.tsx

```diff
@@ -173,24 +173,56 @@ export function jsonToAnalyses(data: Array<any>): Analysis[] {
                 row.state = null;
                 break;
         }
-        return { ...row, selected: false } as Analysis;
+        return { ...row } as Analysis;
     });
     return rows;
 }
 
+/**
+ * Get the index of a material-table row.
+ * If a material-table row is not provided, return null.
+ */
+export function getRowIndex(row: any): number | null {
+    try {
+        let index: number = row.tableData.id;
+        return index;
+    } catch (error) {
+        console.error(error);
+        return null;
+    }
+}
+
+/**
+ * Return whether this material-table row is checked / selected.
+ * If it is not a material-table row, return null.
+ */
+export function getIsRowSelected(row: any): boolean | null {
+    try {
+        let isChecked: boolean = row.tableData.checked;
+        if (isChecked === undefined)
+            isChecked = false;
+        return isChecked;
+    } catch (error) {
+        console.error(error);
+        return null;
+    }
+}
+
 /*****   COMPONENTS   *****/
 
 /**
  * Returns a simple Typography JSX element for displaying "title: value".
  */
-export function FieldDisplay(props: { title: string, value: string | number | null }) {
+export function FieldDisplay(props: { title: string, value?: string[] | string | number | null, variant?: Variant }) {
     let val: string;
     if (typeof props.value === 'number')
         val = props.value.toString();
     else if (typeof props.value === 'string')
         val = props.value;
+    else if (typeof props.value === 'object' && props.value?.length !== undefined)
```

Comment:
Use [Array.isArray](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray).

I think given the context that this value is used in to avoid `null` or `undefined` being displayed, you only need to care about the `null` and array cases since JSX will do the number to string for you below.